python/pyspark/sql/utils.py

```diff
@@ -25,7 +25,12 @@ def __init__(self, desc, stackTrace, cause=None):
         self.cause = convert_exception(cause) if cause is not None else None
 
     def __str__(self):
-        return repr(self.desc)
+        desc = self.desc
+        # encode unicode instance for python2 for human readable description
+        if sys.version_info.major < 3 and isinstance(desc, unicode):
+            return str(desc.encode('utf-8'))
```

Comment:
Let's remove this once we drop Python 2, which will be right away after Spark 3.0 release.

Reply:
OK.